File: shop-app-server/src/main/java/com/medochek/shopapp/domain/Basket.java
package com.medochek.shopapp.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;

@Table
@Entity
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Basket {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @OneToMany(mappedBy = "basket")
    private List<BasketRow> basketRowList;

    @Override
    public String toString() {
        return "Basket{" +
                "id=" + id +
                ", rows=" + basketRowList +
                '}';
    }
}

File: shop-app-web-user/src/main/web/src/app/models/basket.ts
import {BasketRow} from "./basketRow";
import {BasketRowList} from "./basketRowList";

export class Basket {
    private _id: number;
    private _basketRows: BasketRowList;

    constructor(data: any) {
        this._id = data.id ? data.id : null;
        this._basketRows = data.basketRowList  ? new BasketRowList(data.basketRowList) : new BasketRowList({});
    }

    get id(): number {
        return this._id;
    }

    set id(value: number) {
        this._id = value;
    }


    get basketRows(): BasketRowList {
        return this._basketRows;
    }

    set basketRows(value: BasketRowList) {
        this._basketRows = value;
    }
}